drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -265,3 +265,33 @@ int intel_adsp_hda_dma_init(const struct device *dev)
 
 	return 0;
 }
+
+/* DGMBS align value */
+#define HDA_DMA_BUFFER_ALIGNMENT		0x20
```

Comment:
@teburd we should move all of this to dtsi or maybe remove every alignment from dtsi and stick with c defines?
The current mix is probably not looking good

Reply:
The nice part of having these things in device tree, if you use static DEVICE_DT_GET() style macros to get the device, you can also statically get the attributes. And statically setup buffers using those attributes. That's really nice from my perspective.

Those properties aren't available dynamically though. It makes sense, for sof especially, to have the get_attribute() matching up with the existing DMA API there. This could easily return the attributes assigned from DT instead of C defines and you maybe get the best of both.

Reply:
I'm not sure I agree that you would need to store them in a structure, you can get the DT properties just like you have here in the function body without storing them somewhere.

The bigger issue seems to be needing to replicate the properties for each node which I agree is a chore. I'm discussing with @galak to sort out how to avoid that particular issue and how best to do it.

We need these properties of the hardware to be accessible at compile time. While SoF uses dynamic memory allocation in its dai-zephyr.c code, Zephyr tests do not. Generally speaking, Zephyr code and the code that builds upon it wants to know things at compile time which makes sense. The information is there in device tree to do so.

I agree that dma_get_attribute is useful and should be added. What I don't like is that these hardware attributes are now tucked away in a .c file unusable by test code that exists today in Zephyr.